### DISCUSSION TITLE
Build does not work on the latest stable version of NixOS

### DISCUSSION THREAD

**Discussion by pinage404** [90-008/nix-cargo-integration#95] [Q&A] [answered]
Hello,

First of all, thanks for this great tool, which makes it easy to package a Rust project with Nix :slightly_smiling_face: 

The current version of Nix in the latest stable release of NixOS is 2.3

```
nix-build --version
nix-build (Nix) 2.3.16
```

Using this version of Nix, the build fails

```
nix-build --show-trace
warning: unknown setting 'experimental-features'
error: while evaluating the attribute 'cargoArtifacts' of the derivation 'hello-0.0.0' at /nix/store/ny7x54058ki7k1d3ny0jx5h6kw926mxv-source/pkgs/stdenv/generic/make-derivation.nix:278:7:
while evaluating the attribute 'src' of the derivation 'hello-deps-0.0.0' at /nix/store/ny7x54058ki7k1d3ny0jx5h6kw926mxv-source/pkgs/stdenv/generic/make-derivation.nix:278:7:
while evaluating the attribute 'buildCommand' of the derivation 'dummy-src' at /nix/store/ny7x54058ki7k1d3ny0jx5h6kw926mxv-source/pkgs/stdenv/generic/make-derivation.nix:278:7:
while evaluating anonymous function at /nix/store/yi2qhdxw7hrj73l5aj5xdyhsvp576ws6-source/lib/findCargoFiles.nix:4:1, called from /nix/store/yi2qhdxw7hrj73l5aj5xdyhsvp576ws6-source/lib/mkDummySrc.nix:36:12:
attribute 'groupBy' missing, at /nix/store/yi2qhdxw7hrj73l5aj5xdyhsvp576ws6-source/lib/findCargoFiles.nix:32:13
```

I [setup a test project](https://gitlab.com/pinage404/nix_rust_nix_legacy_broken) to play with it, you can try in your browser with the button below

[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://gitlab.com/pinage404/nix_rust_nix_legacy_broken)

I would like to have backward compatibility support

Without updating Nix from 2.3 to the latest version, how can I fix this error?

**Comment by 90-008** [comment, marked as answer]
Why do you need to use an old Nix version? `nci` won't really work with a Nix version that old, at least I would recommend 2.4 or above. If you don't want to update your system Nix version, you can also try entering a shell with `nix-shell -p nixUnstable` and use it like that, and it *should* work.

**Comment by pinage404** [comment]
> Why do you need to use an old Nix version?

I haven't updated Nix on the whole system, so I guess other people haven't either

i will try with `nix-shell -p nixUnstable`
